### ISSUE TITLE
AbsoluteLayout to support positioning from right and bottom

### ISSUE THREAD

**Issue by tuqire** [NativeScript/NativeScript#5591]
Hi,

This is a little bit of a follow on from #5589. Is there anything in the pipeline to support absolute positioning from the right or bottom?

**Comment by tsonevn** [comment]
Hi @tuqire,
First of all, thank you for your interest in NativeScript.
Regarding the issue, could you provide info about your use-case and what you are trying to achieve as a final result?

**Comment by tsonevn** [comment]
HI @tuqire,
As a temporary solution, you could use [StackLayout](https://docs.nativescript.org/ui/layout-containers#sample-horizontal-alignment-of-children) and set up `horizontalAlignment` to `right` to all components, which you need to be right aligned.

**Comment by tuqire** [comment]
that actually makes a lot of sense, thank you.

**Comment by hamdiwanis** [comment]
That will be great to have as in previous app i needed to position an element fixed at the bottom of the screen with the content under it scroll normally, i used an absolute layout and as no bottom property on it i had to use the top property and calculate screen height myself and keep watching for orientation change, so i bottom property would have made it much easier. thanks

**Comment by mohammadrafigh** [comment]
there are many issues closed here and some stackoverflow and forum discussions on it since 2015. I think implementing "bottom" and "right" would be much better than closing all of these issues with some existing workarounds.

**Comment by rainder** [comment]
So how do you propagate events to the parent view? I want to position a huge main button at the bottom of the screen but on top of the other component. How do I make so the component under StackLayout or any other layout would still react to events?

**Comment by SephReed** [comment]
So far the only solution I've found is to put a big fat timeout (> 200 ms) on a view, then hope the parent AbsoluteLayout is done rendering, then get its height and subtract the height of the child (which hopefully is also done rendering), and set top to that.  It's actually surprisingly impossible to have something positioned on the bottom with content under it.

**Comment by NathanaelA** [comment]
For those who don't know -- you can use a GridLayout to position items at the bottom of the screen under other items at the bottom of the screen.  ;-)   
```
<GridLayout rows="*,auto">
   <ItemTakingFullScreen rowSpan="2"/>
   <ItemShownUnder row="1"/>
   <ItemShownAbove row="1">
</GridLayout>
```
You can also even replace ItemTakingFullScreen with a ScrollView and then the scrollview content will go under the ItemShownAbove content.

**Comment by SephReed** [comment]
WOAH!  that's how grid layout works!?

I was thinking old html table style.  I'm sorry, and thank you for clarifying.  There's like ... 3 stack overflow posts I'm going to drop this answer in on.

**Comment by SephReed** [comment]
In the api for [`GridLayout`](https://docs.nativescript.org/api-reference/classes/_ui_layouts_grid_layout_.gridlayout) I can find no property named "rows" or method which takes a string argument for creating rows.  Is this function not public facing?  If so, is there any simple way to create an [`ItemSpec`](https://docs.nativescript.org/api-reference/classes/_ui_layouts_grid_layout_.itemspec) from these strings?

**Comment by NathanaelA** [comment]
@SephReed - GridLayout is my favorite layout.   Because you can do any number of rows and columns, and you can mix and match the sizes (like use `20%`, `*`, `30px`, `5dp`) and then you can use colSpan and rowSpan to position things across rows/cols , and everything stacks in logical order; it really is the swiss army knife of layouts.   I rarely use another layout.  :grinning:   You might try https://www.nslayouts.com/ starting at lesson 11 to learning about the different options that GridLayouts has.

**Comment by SephReed** [comment]
Found it:
https://github.com/NativeScript/NativeScript/blob/e0bddd77485bc8054ab80bb325679d69c086264d/tns-core-modules/ui/layouts/grid-layout/grid-layout-common.ts#L44

Appears not to be a public facing function.  I can just copy and paste.  Might be nice to add it to GridLayout as a public function though.

**Comment by NathanaelA** [comment]
Yeah, the API documentation doesn't do a good job of teaching it.   `rows` & `columns` (with the `s`) is used inside the GridLayout definition.   `row` `column` (without the `s`) is used in the control embedded in the GridLayout.   Not setting a row/column in a control defaults it to 0.   
So `<GridLayout rows="*,*,*,20"/>` would mean that there are 4 rows, the 4th row is taking 20dp units.  The other 3 rows are splitting the remaining space equally in 3rds.  (Same applies to columns)

**Comment by SephReed** [comment]
Oh, the documentation for xml is great.  But I'm not using any xml.  I did figure it out though.

```
// grid helper
/*********************
 * Grid
 */
export interface IGridLayout extends IViewProps {
	columns?: string;
	rows?: string;
}
export function grid(
	classOrProps?: string | IGridLayout,
	propsOrInnards?: IGridLayout | Innards,
	innardsOrNot?: Innards,
) {
	const out = new GridLayout();
	const { props } = _layoutHelper<IGridLayout>(out, classOrProps, propsOrInnards, innardsOrNot);
	if (props.columns) {
		(out as any).columns = props.columns;
	}
	if (props.rows) {
		(out as any).rows = props.rows;
	}
	return out;
}

// in use
return grid("MainAppView", { rows: "*,auto" }, [
	this.viewPort.render(),
	this.bottomNav.render(),
]);
```

**Comment by Arystosedes** [comment]
Any news? It could be very helpful in many cases.

**Comment by prajilshresthaiw** [comment]
Any news?

**Comment by neversaynever44** [comment]
any news?

**Comment by miladfm** [comment]
Any news?

**Comment by nicolaric** [comment]
any news?

**Comment by VaidasDatenis** [comment]
any news?

**Comment by giuseppemorale** [comment]
any news?
